map_server/src/main.cpp

```diff
@@ -67,156 +68,218 @@ class MapServer
+      map_pub = n.advertise<nav_msgs::OccupancyGrid>("map", 1, true);
+      
       deprecated = (res != 0);
       if (!deprecated) {
-        //mapfname = fname + ".pgm";
-        //std::ifstream fin((fname + ".yaml").c_str());
-        std::ifstream fin(fname.c_str());
-        if (fin.fail()) {
-          ROS_ERROR("Map_server could not open %s.", fname.c_str());
+        if (!loadMapFromYaml(fname))
+        {
           exit(-1);
         }
-#ifdef HAVE_YAMLCPP_GT_0_5_0
-        // The document loading process changed in yaml-cpp 0.5.
-        YAML::Node doc = YAML::Load(fin);
-#else
-        YAML::Parser parser(fin);
-        YAML::Node doc;
-        parser.GetNextDocument(doc);
-#endif
-        try {
-          doc["resolution"] >> res;
-        } catch (YAML::InvalidScalar &) {
-          ROS_ERROR("The map does not contain a resolution tag or it is invalid.");
-          exit(-1);
-        }
-        try {
-          doc["negate"] >> negate;
-        } catch (YAML::InvalidScalar &) {
-          ROS_ERROR("The map does not contain a negate tag or it is invalid.");
-          exit(-1);
-        }
-        try {
-          doc["occupied_thresh"] >> occ_th;
-        } catch (YAML::InvalidScalar &) {
-          ROS_ERROR("The map does not contain an occupied_thresh tag or it is invalid.");
-          exit(-1);
-        }
-        try {
-          doc["free_thresh"] >> free_th;
-        } catch (YAML::InvalidScalar &) {
-          ROS_ERROR("The map does not contain a free_thresh tag or it is invalid.");
+      } else {
+        if (!loadMapFromParams(fname, res))
+        {
           exit(-1);
         }
-        try {
-          std::string modeS = "";
-          doc["mode"] >> modeS;
+      }
+    }
 
-          if(modeS=="trinary")
-            mode = TRINARY;
-          else if(modeS=="scale")
-            mode = SCALE;
-          else if(modeS=="raw")
-            mode = RAW;
-          else{
-            ROS_ERROR("Invalid mode tag \"%s\".", modeS.c_str());
-            exit(-1);
-          }
-        } catch (YAML::Exception &) {
-          ROS_DEBUG("The map does not contain a mode tag or it is invalid... assuming Trinary");
-          mode = TRINARY;
-        }
-        try {
-          doc["origin"][0] >> origin[0];
-          doc["origin"][1] >> origin[1];
-          doc["origin"][2] >> origin[2];
-        } catch (YAML::InvalidScalar &) {
-          ROS_ERROR("The map does not contain an origin tag or it is invalid.");
-          exit(-1);
-        }
-        try {
-          doc["image"] >> mapfname;
-          // TODO: make this path-handling more robust
-          if(mapfname.size() == 0)
-          {
-            ROS_ERROR("The image tag cannot be an empty string.");
-            exit(-1);
-          }
+  private:
+    ros::NodeHandle n;
+    ros::Publisher map_pub;
+    ros::Publisher metadata_pub;
+    ros::ServiceServer service;
+    ros::ServiceServer change_map_srv_;
+    bool deprecated;
+    std::string frame_id_;
```

Comment:
should really be more consistent in naming here - member variable should all have the trailing underscore_